skfda/inference/hotelling/_hotelling.py

```diff
@@ -85,7 +85,7 @@ def hotelling_t2(
     n = n1 + n2  # Size of full sample
     m = fd1.mean() - fd2.mean()  # Delta mean
 
-    if isinstance(fd1, FDataBasis):
+    if isinstance(fd1, FDataBasis) and isinstance(fd2, FDataBasis):
```

Comment:
As of now, `cov()` returns and `FData` object and `cov(s,t)` returns the evaluation of the covariance in the grid formed by the cartesian product of s and t. Thus, there is no direct method for getting the data matrix or the coefficients matrix of the covariance function.